.github/workflows/trigger-gitlab.yml

```diff
@@ -88,3 +91,33 @@ jobs:
           else
             git push -f ci 
           fi
+
+      - name: Trigger GitLab nightly pipeline against this PR
+        if: ${{ env.GITLAB_TOKEN }} && ${{ steps.pr_date.outputs.pr_branch }}
+        run: |
+            # osbuild-composer
+            PROJECT_ID=34771166
+
+            # TODO: create if it doesn't exist (subsequent pushes)
+            NEW_SCHEDULE=$(curl --request POST --header "PRIVATE-TOKEN: $GITLAB_TOKEN" \
+                 --form description="9.3 nightly against ${{ steps.pr_date.outputs.pr_branch }}" \
+                 --form ref="${{ steps.pr_date.outputs.pr_branch }}" \
+                 --form cron="0 1 * * 5" --form cron_timezone="UTC" \
+                 --form active="false" \
+                 "https://gitlab.com/api/v4/projects/$PROJECT_ID/pipeline_schedules")
+            SCHEDULE_ID=$(jq .id <<< $NEW_SCHEDULE)
+
+            # adjust variables for new schedule
```

Comment:
Isn't this potentially racy? You can pass variable in the trigger I think: https://docs.gitlab.com/ee/ci/triggers/#pass-cicd-variables-in-the-api-call

Reply:
> You can pass variable in the trigger I think: https://docs.gitlab.com/ee/ci/triggers/#pass-cicd-variables-in-the-api-call

I will update for this. 

> Isn't this potentially racy?

However racy with what ?

Reply:
I think I see what you mean here but I don't see how I can specify the pipeline I want to trigger. There's PROJECT_ID and TOKEN, which is used for authentication. I don't see where I can specify the actual pipeline that I want to trigger.

Reply:
Ah, I see, we are running this as a scheduled pipeline. In that case, your method is probably the only viable one, since you cannot pass extra variables to the `play` route: https://docs.gitlab.com/ee/api/pipeline_schedules.html#run-a-scheduled-pipeline-immediately

The alternative would to run it just a plain pipeline and not a scheduled. I'm not sure, though, how much would that change all our scripts.

Reply:
> The alternative would to run it just a plain pipeline and not a scheduled.

The question around changes aside, how does trigger specify which pipeline to execute? I assume it will execute all that are defined, no ?

Reply:
Indeed, we would have to make them running conditionally based on a variable.

Reply:
I've discovered that I can actually pass `--form "variables[CI_PIPELINE_SOURCE]=schedule"` as part of the POST payload. It triggers the nightly pipeline b/c we have conditionals on the CI_PIPELINE_SOURCE variable. However GitLab's UI doesn't show the pipeline as "Scheduled" b/c it isn't. Everything else seems to be as expected:
https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/pipelines/920086961